### PR TITLE
Defined Organizations events, states, rejections and contract

### PR DESCRIPTION
**Pull request by umbreak**
Fixes https://github.com/BlueBrain/nexus/issues/1489

### CODE REVIEW

File: delta/sdk/src/main/scala/ch/epfl/bluebrain/nexus/delta/sdk/model/organizations/OrganizationState.scala [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+import ch.epfl.bluebrain.nexus.delta.sdk.model.Identity.Subject
+import ch.epfl.bluebrain.nexus.delta.sdk.model.ResourceRef.Latest
+import ch.epfl.bluebrain.nexus.delta.sdk.model.{Label, ResourceF, ResourceRef}
+import org.apache.jena.iri.IRI
+
+/**
+  * Enumeration of organization states.
+  */
+sealed trait OrganizationState extends Product with Serializable {
+
```
Comment (bogdanromanx) [inline delta/sdk/src/main/scala/ch/epfl/bluebrain/nexus/delta/sdk/model/organizations/OrganizationState.scala:17]: add label, uuid and deprecated

File: delta/sdk/src/main/scala/ch/epfl/bluebrain/nexus/delta/sdk/model/search/Pagination.scala [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+  *
+  * @param searchAfter  [[Json]] to use as ElasticSearch `search_after` field. Should be directly taken from `sort` field
+  *                     in the search results.
+  * @param size         the maximum number of results per page
+  */
+final case class SearchAfterPagination(searchAfter: Json, size: Int) extends Pagination
+
+object Pagination {
+
+  def apply(size: Int): Pagination                    = FromPagination(0, size)
```
Comment (bogdanromanx) [inline delta/sdk/src/main/scala/ch/epfl/bluebrain/nexus/delta/sdk/model/search/Pagination.scala:37]: add docs

File: delta/sdk/src/main/scala/ch/epfl/bluebrain/nexus/delta/sdk/model/search/SearchParams.scala [hunk 3]
```diff
@@ -0,0 +1,65 @@
+package ch.epfl.bluebrain.nexus.delta.sdk.model.search
+
+import ch.epfl.bluebrain.nexus.delta.sdk.model.{Label, ResourceRef}
+import org.apache.jena.iri.IRI
+
+sealed trait SearchParams {
```
Comment (bogdanromanx) [inline delta/sdk/src/main/scala/ch/epfl/bluebrain/nexus/delta/sdk/model/search/SearchParams.scala:6]: add docs

File: delta/sdk/src/main/scala/ch/epfl/bluebrain/nexus/delta/sdk/model/search/SearchParams.scala [hunk 4]
```diff
@@ -0,0 +1,65 @@
...
+  def types: Set[IRI]
+  def schemas: Set[ResourceRef]
+}
+
+object SearchParams {
+
+  /**
+    * Search parameters to filter realm resources.
+    *
+   * @param deprecated the optional deprecation status of the realm resources
```
Comment (bogdanromanx) [inline delta/sdk/src/main/scala/ch/epfl/bluebrain/nexus/delta/sdk/model/search/SearchParams.scala:20]: indent issue

File: delta/sdk/src/main/scala/ch/epfl/bluebrain/nexus/delta/sdk/model/search/SearchParams.scala [hunk 5]
```diff
@@ -0,0 +1,65 @@
...
+    * @param rev        the optional revision of the realm resources
+    * @param createdBy  the optional identity id who created the realm resource
+    * @param updatedBy  the optional identity id who updated the realm resource
+    * @param types      the collection of types of the realm resources
+    * @param schemas    the collection of schemas of the realm resources
+    */
+  final case class RealmSearchParams(
+      deprecated: Option[Boolean] = None,
+      rev: Option[Long] = None,
```
Comment (bogdanromanx) [inline delta/sdk/src/main/scala/ch/epfl/bluebrain/nexus/delta/sdk/model/search/SearchParams.scala:29]: I don't think rev makes sense... nor types and schemas, to be honest

File: delta/sdk/src/main/scala/ch/epfl/bluebrain/nexus/delta/sdk/model/search/SearchParams.scala [hunk 6]
```diff
@@ -0,0 +1,65 @@
...
+  ) extends SearchParams
+
+  object RealmSearchParams {
+    final val none: RealmSearchParams = RealmSearchParams()
+  }
+
+  /**
+    * Search parameters to filter organization resources.
+    *
+   * @param label      the optional label of the organization resources
```
Comment (bogdanromanx) [inline delta/sdk/src/main/scala/ch/epfl/bluebrain/nexus/delta/sdk/model/search/SearchParams.scala:43]: same as above, for organizations we don't need label, rev, types and schemas.

File: delta/sdk/src/main/scala/ch/epfl/bluebrain/nexus/delta/sdk/model/search/SearchParams.scala [hunk 7]
```diff
@@ -0,0 +1,65 @@
...
+      deprecated: Option[Boolean] = None,
+      rev: Option[Long] = None,
+      createdBy: Option[IRI] = None,
+      updatedBy: Option[IRI] = None,
+      types: Set[IRI] = Set.empty,
+      schemas: Set[ResourceRef] = Set.empty
+  ) extends SearchParams
+
+  object OrganizationSearchParams {
+    final val none: OrganizationSearchParams = OrganizationSearchParams()
```
Comment (bogdanromanx) [inline delta/sdk/src/main/scala/ch/epfl/bluebrain/nexus/delta/sdk/model/search/SearchParams.scala:62]: add docs

File: delta/sdk/src/main/scala/ch/epfl/bluebrain/nexus/delta/sdk/model/search/SearchResults.scala [hunk 8]
```diff
@@ -0,0 +1,102 @@
...
+        fa match {
+          case sqr: ScoredSearchResults[A]   => sqr.map(f)
+          case uqr: UnscoredSearchResults[A] => uqr.map(f)
+        }
+    }
+
+  /**
+    * Constructs an [[ScoredSearchResults]]
+    *
+   * @param total      the total number of results
```
Comment (bogdanromanx) [inline delta/sdk/src/main/scala/ch/epfl/bluebrain/nexus/delta/sdk/model/search/SearchResults.scala:87]: indenting issues :)

**Review by bogdanromanx — CHANGES_REQUESTED** [review]

File: delta/sdk/src/main/scala/ch/epfl/bluebrain/nexus/delta/sdk/model/organizations/OrganizationState.scala (hunk 1 above)
Comment (umbreak) [reply delta/sdk/src/main/scala/ch/epfl/bluebrain/nexus/delta/sdk/model/organizations/OrganizationState.scala:17]: There is no label, uuid or deprecation flag for Initial state. So how this is helpful? I would have to add
```
def label: Option[Label]
def uuid: Option[UUID]
```

and for what is useful?

File: delta/sdk/src/main/scala/ch/epfl/bluebrain/nexus/delta/sdk/model/search/SearchParams.scala (hunk 5 above)
Comment (umbreak) [reply delta/sdk/src/main/scala/ch/epfl/bluebrain/nexus/delta/sdk/model/search/SearchParams.scala:29]: rev make sense, why not? types and schemas I can remove

Comment (bogdanromanx) [reply delta/sdk/src/main/scala/ch/epfl/bluebrain/nexus/delta/sdk/model/search/SearchParams.scala:29]: I don't see how: get all the realms at revision 2 would make any sense... :)

File: delta/sdk/src/main/scala/ch/epfl/bluebrain/nexus/delta/sdk/model/search/SearchParams.scala (hunk 6 above)
Comment (umbreak) [reply delta/sdk/src/main/scala/ch/epfl/bluebrain/nexus/delta/sdk/model/search/SearchParams.scala:43]: rev yes

**Review by imsdu — APPROVED** [review]

**Review by bogdanromanx — CHANGES_REQUESTED** [review]

**Review by bogdanromanx — APPROVED** [review]

**Pull request merged**